includes/api/class-wc-admin-rest-reports-stock-controller.php

```diff
@@ -0,0 +1,354 @@
+	 */
+	public function get_items( $request ) {
+		$query_args    = $this->prepare_reports_query( $request );
+		$query_results = $this->get_products( $query_args );
+
+		$objects = array();
+		foreach ( $query_results['objects'] as $object ) {
+			$data      = $this->prepare_item_for_response( $object, $request );
+			$objects[] = $this->prepare_response_for_collection( $data );
+		}
+
+		$page      = (int) $query_args['paged'];
+		$max_pages = $query_results['pages'];
+
+		$response = rest_ensure_response( $objects );
+		$response->header( 'X-WP-Total', $query_results['total'] );
+		$response->header( 'X-WP-TotalPages', (int) $max_pages );
+
+		$base = add_query_arg( $request->get_query_params(), rest_url( sprintf( '/%s/%s', $this->namespace, $this->rest_base ) ) );
+
+		if ( $page > 1 ) {
+			$prev_page = $page - 1;
+			if ( $prev_page > $max_pages ) {
+				$prev_page = $max_pages;
+			}
+			$prev_link = add_query_arg( 'page', $prev_page, $base );
+			$response->link_header( 'prev', $prev_link );
+		}
+		if ( $max_pages > $page ) {
+			$next_page = $page + 1;
+			$next_link = add_query_arg( 'page', $next_page, $base );
+			$response->link_header( 'next', $next_link );
+		}
+
+		return $response;
+	}
+
+	/**
+	 * Prepare a report object for serialization.
+	 *
+	 * @param  WC_Product      $product  Report data.
+	 * @param  WP_REST_Request $request Request object.
+	 * @return WP_REST_Response
+	 */
+	public function prepare_item_for_response( $product, $request ) {
+		$data = array(
+			'id'             => $product->get_id(),
+			'parent_id'      => $product->get_parent_id(),
+			'sku'            => $product->get_sku(),
+			'stock_status'   => $product->get_stock_status(),
+			'stock_quantity' => (float) $product->get_stock_quantity(),
+		);
+
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->add_additional_fields_to_object( $data, $request );
+		$data    = $this->filter_response_by_context( $data, $context );
+
+		// Wrap the data in a response object.
+		$response = rest_ensure_response( $data );
+		$response->add_links( $this->prepare_links( $product ) );
+
+		/**
+		 * Filter a report returned from the API.
+		 *
+		 * Allows modification of the report data right before it is returned.
+		 *
+		 * @param WP_REST_Response $response The response object.
+		 * @param WC_Product       $product   The original product object.
+		 * @param WP_REST_Request  $request  Request used to generate the response.
+		 */
+		return apply_filters( 'woocommerce_rest_prepare_report_stock', $response, $product, $request );
+	}
+
+	/**
+	 * Prepare links for the request.
+	 *
+	 * @param  WC_Product $product Object data.
+	 * @return array
+	 */
+	protected function prepare_links( $product ) {
+		$links = array(
+			'product' => array(
+				'href' => rest_url( sprintf( '/%s/products/%d', $this->namespace, $product->get_parent_id() ? $product->get_parent_id() : $product->get_id() ) ),
+			),
+		);
```

Comment:
Just thinking out loud here, but should this maybe return `https://local.wordpress.test/wp-json/wc/v3/products/XYZ/variations/ABC` for variations?